src/pages/account/profile/index.tsx

```diff
@@ -27,27 +27,36 @@ import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
 const ProfilePage: React.FC = () => {
   const router = useRouter();
 
-  const { data, error } = useSWR<
-    {
-      userProfile: Profile;
-      userAddresses: Address[];
-    },
+  const { data: userProfile, error: userProfileError } = useSWR<
+    Profile,
     SWRError
-  >(`/api/profile/get-user-profile-and-addresses`);
+  >(`/api/profile/user-profile`);
 
-  if (error) {
-    if (error.statusCode === HttpStatus.NOT_FOUND) {
+  if (userProfileError) {
+    if (userProfileError.statusCode === HttpStatus.NOT_FOUND) {
       router && router.push("/account/profile/user-profile/new");
```

Comment:
```suggestion
      router && router.replace("/account/profile/user-profile/new");
```
Because otherwise it is added to the history and if the user goes back after the redirect it is redirect again

Reply:
Very good point 👍